### ISSUE TITLE
Openstack API error messages are not propagated to the user

### ISSUE THREAD

**Issue by ruoso** [hashicorp/terraform#11868]
Running with terraform 0.8.6, I had a configuration that would just give me the output:

```
* openstack_compute_instance_v2.terraform: Error creating OpenStack server: Invalid request due to incorrect syntax or missing required parameters.
```

Even enabling trace wouldn't give me any more useful output to indicate what was wrong. After a lot of suffering I finally decided to run terraform with mitmproxy to be able to see if there was anything useful in the requests/responses themselves, and it turns out that the openstack server was returning a very useful error message that allowed me to solve the problem right away (flavor disk too small for image).

Please propagate the errors from the openstack API to the user, or at least add debugging output with the responses such that we have some hope of troubleshooting.

**Comment by jtopjian** [comment]
@ruoso Agreed. This is an issue with the underlying OpenStack library that Terraform uses. This is a known issue and will hopefully be resolved soon. I'll leave this Issue open as a placeholder and will update it if there's any news.

**Comment by ruoso** [comment]
Maybe a short term answer is to have gophercloud just have an env variable for dumping the json for the requests and responses... But I guess that'd be an issue on that repo, not here.

**Comment by jtopjian** [comment]
That should be possible to do within Terraform. I will investigate this.

**Comment by jtopjian** [comment]
@ruoso I've just submitted #12089 which enables this functionality. Feel free to test it out and let me know if you have any feedback :)

**Comment by stack72** [comment]
Closed via #12089
